### ISSUE TITLE
`xmp: error no XMP Tag found` even though multiple XMP tags are found

### ISSUE THREAD

**Issue by mholt** [evanoberholster/imagemeta#45]
I'm using the example from the readme on a JPEG file:

```
m, err := imagemeta.Parse(f)
if err != nil {
panic(err)
}
fmt.Println(m.Exif())
fmt.Println(m.Xmp())
```

but I get an error of `xmp: error no XMP Tag found`, even though this file has two XMP sections.

I've cloned the repo and am trying to get a grasp for how the readers work, but there's lots of sectioning and ReadAt() and it's a little hard for me to follow.

Any ideas or troubleshooting tips?

Thanks for this package!

**Comment by mholt** [comment]
Additionally, I have a `.heif` file that has an XMP section:

```
$ grep --binary --text --byte-offset '<x:xmpmeta' 20220423_085935.heif 
974650:debuginfo<x:xmpmeta xmlns:x="adobe:ns:meta/" x:xmptk="Adobe XMP Core 5.1.0-jc003">
```

But the same code yields an error: `xmp: error no XMP Tag found`

**Comment by evanoberholster** [comment]
@mholt, any chance you could share those images with me. That would assist in debugging. 

This library is definitely no perfect and written with the goal of performance, so there are likely cases where it needs to be tweaked.

**Comment by mholt** [comment]
@evanoberholster Absolutely! Sorry, that would have been the obvious helpful thing to do :)

Here's one: https://drive.google.com/file/d/1mmDRBPfxDenvCQz0wezygyGhcWdoNJHk/view?usp=sharing

One thing I have noted in my debugging is that there are XMP fields that are longer than 1.5 KB; notably, I think the MakerNotes on this picture are about 70-80 KB. I've seen some up to 100 KB.

When I raise maximum value and buffer sizes, I am also seeing that `Peek()` in the `readAttributeValue` loop somehow stops returning more than about 60 KB at a time, resulting in never finding the closing delimiter. This picture should have makernotes going from 9830 to 78430, but the read buffer will never be large enough to see the closing quotes at 78430.

One thing I also tried in my debugging is to change the signature of `Xmp()` from returning `xmp.XMP` to `[]xmp.XMP`, since a file may have multiple XMP documents. (In this file's case, I suspect the second one is for the embedded mp4 file, i.e. motion photo. But my application can know to use only the first or the second depending on what it's doing.)

Thank you so much for the reply! I've been trying a few other libraries yours readme mentions before I found this one, and this one looks the most maintained, so I really appreciate that.

**Comment by evanoberholster** [comment]
@mholt. Thanks for sending me that jpg file. It was eye opening. I had not come across a similar file layout as in this image. When I am debugging a file structure I normally use ```./exiftool -htmldump image.jpg > image.jpg.html``` as it gives a good idea of the structure. 

In the JPEG that you posted it appears to have APP0 JFIF tags which I hadn't dealt with in my library before. c89196d

Most of the XMP files that I work with come from Lightroom and therefore I have used a smaller buffer (1.5KB). Thank you for the suggestion, I will look for more XMP files to doing testing with.

This go library is definitely undergoing some refactoring and rewriting that you will see under the develop branch. Thank you for the suggestion of returning an ```[]xmp.XMP``` instead of an ```xmp.XMP```

I will also take a look at the HEIC image in the next few days.

**Comment by mholt** [comment]
@evanoberholster That's awesome, thank you! I hadn't noticed the `develop` branch.

I'm kind of new to parsing metadata, so I've learned a ton this week. And wow, `exiftool -htmldump` is truly useful, thanks for the tip.

Of the three metadata libraries I've tried so far -- and all of them have struggled with the first few images I've tried -- this is the only one that appears to be maintained. So thank you, thank you again for that. It's a breath of fresh air.

Can I possibly sponsor you or donate?

Whatever I can do to test changes or help with development, please let me know!

**Comment by mholt** [comment]
@evanoberholster Sounds good to me. Thanks for explaining that -- I totally understand not having the desire to support proprietary fields. Can your lib at least extract the contents of the MakerNotes? It doesn't have to decode them. But if it's part of a standard xmp document then I hope we could at least access them for later.

> I will be pushing a large rewrite soon and would appreciate suggestions and testing.

I'm looking forward to trying it! Will it have the ability to detect the file type? Or will I need to call a type-specific function?

**Comment by lordofscripts** [comment]
There is no `imagemeta.Parse()` function nor `m.Xmp()` method.
